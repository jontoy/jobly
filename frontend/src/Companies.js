import React, {useState, useEffect} from 'react';
import JoblyApi from './JoblyApi';
import Searchbox from './Searchbox';
import CompanyList from './CompanyList';

const Companies = () => {

    const [isLoading, setIsLoading] = useState(true);
    const [companies, setCompanies] = useState([]);
    const [searchTerm, setSearchTerm] = useState('');
    useEffect(() => {
      async function getCompanies() {
        let allCompanies = await JoblyApi.getCompanies({search:searchTerm});
        setCompanies(allCompanies);
      }
      getCompanies();
      setIsLoading(false);
    }, [searchTerm]);
    if (isLoading) {
      return <p>Loading &hellip;</p>;
    }

    return (
        <div className="Companies container">
            <Searchbox filterResults={setSearchTerm}/>
            <CompanyList companies={companies}/>
        </div>
    )
}

export default Companies;