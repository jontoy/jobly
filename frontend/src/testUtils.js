import React from "react";
import UserContext from "./userContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs:[]
};

const UserProvider = ({ children, currentUser = demoUser, appliedIds = [] }) => (
  <UserContext.Provider value={{currentUser, appliedIds}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
