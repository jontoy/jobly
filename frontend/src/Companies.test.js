import React from 'react';
import { render } from '@testing-library/react';
import Companies from './Companies';
import {UserProvider} from './testUtils';

it('renders without crashing', () => {
    render(
    // <UserProvider>
        <Companies />
    // </UserProvider>
    );
});
  
it('matches snapshot', () => {
      const {asFragment} = render(<Companies />);
    expect(asFragment()).toMatchSnapshot();
});
