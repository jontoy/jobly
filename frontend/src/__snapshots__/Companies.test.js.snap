// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Companies container"
  >
    <form
      class="Searchbox form-inline mt-3"
    >
      <input
        class="form-control form-control-lg flex-grow-1"
        name="search"
        placeholder="Search..."
        type="search"
        value=""
      />
    </form>
    <ul
      class="CompanyList list-unstyled"
    />
  </div>
</DocumentFragment>
`;
